mmpose/datasets/pipelines/shared_transform.py

```diff
@@ -27,11 +28,18 @@ class ToTensor:
         results (dict): contain all information about training.
     """
 
+    def __init__(self, device='cpu'):
+        self.device = device
+
     def __call__(self, results):
+        to_tensor = (lambda x: torch.from_numpy(x.astype('float32')).permute(
+            2, 0, 1).to(self.device))
         if isinstance(results['img'], (list, tuple)):
-            results['img'] = [F.to_tensor(img) for img in results['img']]
+            results['img'] = [
+                to_tensor(img.astype('float32')) for img in results['img']
+            ]
         else:
-            results['img'] = F.to_tensor(results['img'])
+            results['img'] = to_tensor(results['img'].astype('float32'))
```

Comment:
It seems that `.astype('float32')` is not necessary, because in `to_tensor`, x will first transformed to 'float32'.

Reply:
I'll fix it